src/Components/Categories/CategoriesTable.js

```diff
@@ -0,0 +1,144 @@
+
+library.add(faTrashAlt);
+library.add(faEdit);
+
+export default function CategoriesTable(props) {
+  const renderDateColumn = (dateString) =>
+    formatDate(parseISO(dateString), "PP");
+
+  const [name, setName] = useState("");
+  const [selectedCategory, setSelectedCategory] = useState({});
+
+  const [showCategoryEdit, setShowCategoryEdit] = useState(false);
+  const handleClose = () => {
+    setShowCategoryEdit(false);
+  };
+  const {
+    currentOrganizationStore,
+    currentOrganizationDispatch,
+    organizationClient,
+  } = useCurrentOrganizationContext();
+  const currentOrganizationId =
+    currentOrganizationStore.currentOrganization &&
+    currentOrganizationStore.currentOrganization.id;
+
+  const handleShowCategoryEdit = (selectedCategory) => {
+    setSelectedCategory(selectedCategory);
+    setShowCategoryEdit(true);
+  };
+
+  const handleSubmitEditCategory = ({ newName }, id) => {
+    console.log(id);
+    updateCategory(organizationClient, id, {
+      name: newName,
+      organization_id: currentOrganizationId,
+    })
+      .then((category) => {
+        setName(name);
+        handleClose();
+      })
+      .catch((error) => {
+        console.log("category update error", error);
+      });
+  };
+
+  const handleCancel = (event) => {
+    handleClose();
+  };
+
+  const handleCategoryDelete = (categoryId) => {
+    console.log("deleted!");
+  };
+
+  const columns = useMemo(
+    () => [
+      { Header: "Name", accessor: "name" },
+      {
+        Header: "Date Created",
+        accessor: (row) => renderDateColumn(row.created_at),
+      },
+      {
+        Header: "Last Modified",
+        accessor: (row) => renderDateColumn(row.updated_at),
+      },
+    ],
+    []
+  );
+
+  const categories = useMemo(() => props.categories, [props.categories]);
+  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
+    useTable({ columns, data: categories });
+
+  const header = headerGroups.map((headerGroup) => (
+    <tr {...headerGroup.getHeaderGroupProps()}>
+      {headerGroup.headers.map((column) => (
+        <th {...column.getHeaderProps()}>{column.render("Header")}</th>
+      ))}
+    </tr>
+  ));
+
+  const body = rows.map((row) => {
+    prepareRow(row);
+
+    return (
+      <tr {...row.getRowProps()}>
+        {row.cells.map((cell) => {
+          const renderedCell = (() => {
+            return cell.render("Cell");
+          })();
+
+          return <td {...cell.getCellProps()}>{renderedCell}</td>;
+        })}
+        <FontAwesomeIcon
+          icon={faEdit}
+          style={{
+            color: "black",
+            fontSize: "1.5rem",
+          }}
+          onClick={() => handleShowCategoryEdit(row)}
+        />
+        <FontAwesomeIcon
+          icon={faTrashAlt}
+          style={{
+            color: "black",
+            fontSize: "1.5rem",
+          }}
+          onClick={() => handleCategoryDelete(row.id)}
+        />
```

Comment:
I'm getting this error in the console when we have these icons (which become svgs) as children of the `tr` here:
```
index.js:1 Warning: validateDOMNesting(...): <svg> cannot appear as a child of <tr>.
    in svg (created by FontAwesomeIcon)
    in FontAwesomeIcon (at CategoriesTable.js:109)
    in tr (at CategoriesTable.js:101)
    in tbody (at CategoriesTable.js:133)
    in table (at CategoriesTable.js:131)
    in div (at CategoriesTable.js:130)
    in CategoriesTable (at Categories.js:143)
    in div (at Categories.js:101)
    in Categories (at PrivateRoute.js:14)
    in Route (at PrivateRoute.js:10)
    in PrivateRoute (at App.js:127)
    in Switch (at App.js:70)
    in Router (created by BrowserRouter)
    in BrowserRouter (at App.js:67)
    in div (at App.js:66)
    in App (at src/index.js:19)
    in CurrentOrganizationProvider (at src/index.js:17)
    in CurrentUserProvider (at src/index.js:16)
    in StrictMode (at src/index.js:15)
```

Reply:
I was able to get it to go away by wrapping the icons in their own `<td>`:

```suggestion
        <td>
          <FontAwesomeIcon
            icon={faEdit}
            style={{
              color: "black",
              fontSize: "1.5rem",
            }}
            onClick={() => handleShowCategoryEdit(row)}
          />
          <FontAwesomeIcon
            icon={faTrashAlt}
            style={{
              color: "black",
              fontSize: "1.5rem",
            }}
            onClick={() => handleCategoryDelete(row.id)}
          />
       </td>
```

Reply:
Also, another important thing: accessibility.

These icons are meant to act as buttons, but since they are only a `svg` in the markup, screen readers would not be able to know they're buttons!

A quick fix would be to wrap each icon in a `<button>` tag. This might make the icons look weird visually, but we can fix that with some css!

This blog post has a nice walkthrough: https://www.sarasoueidan.com/blog/accessible-icon-buttons/

Reply:
Oooh, thank you! I think this is probably something to push to after we figure out what's going on with Material-UI - Eli wants us to not use FontAwesome icons, but rather Material-UI ones. I'll add in the td fix.